syntax = "proto3";

package nlp;

message KeywordRequest {
    enum MethodType {
        TFIDF = 0;
        TextRank = 1;
    }

    MethodType method = 1;

    string sentence = 2;
    int64 topK = 3;
    repeated string allowPOS = 4;
}

message KeywordResponse {
    message Keyword {
        string word = 1;
        double weight = 2;
        string pos = 3;
    }
    repeated Keyword keywords = 1;
}

service NLP {
    rpc KeywordExtract(KeywordRequest) returns (KeywordResponse);

    rpc KeywordExtractStream(stream KeywordRequest) returns (KeywordResponse);
}
